tgui/packages/tgui/interfaces/common/AtmosHandbook.tsx

```diff
@@ -0,0 +1,242 @@
+          onChange={(keyword) =>
+            setActiveGasId(
+              gasTextBook.find((gas) =>
+                gas.name.toLowerCase().startsWith(keyword.toLocaleLowerCase())
+              )?.id || '')}
+        />
+      }
+      fill
+      scrollable>
+      {relevantGas && (
+        <>
+          <Box mb={'0.5em'}>{relevantGas.description}</Box>
+          <Box mb={'0.5em'}>
+            {'Specific heat: ' + relevantGas.specific_heat + ' Joule/KelvinMol'}
+          </Box>
+          <Box mb={'0.5em'}>{'Relevant Reactions:'}</Box>
+          {Object.keys(relevantGas.reactions).map((reaction_id) => (
+            <Flex key={reaction_id} mb="0.5em">
+              <Flex.Item
+                className="AtmosHandbook__link-content"
+                shrink
+                onClick={() => setActiveReactionId(reaction_id)}>
+                {relevantGas.reactions[reaction_id]}
+              </Flex.Item>
+            </Flex>
+          ))}
+        </>
+      )}
+    </Section>
+  );
+};
+
+const ReactionHandbook = (props: { reactionTextBook: Reaction[] }, context) => {
+  const { act, data } = useBackend(context);
+  const { reactionTextBook } = props;
+  const [activeGasId, setActiveGasId] = useSharedState(
+    context,
+    'activeGasId',
+    ''
+  );
+  const [activeReactionId, setActiveReactionId] = useSharedState(
+    context,
+    'activeReactionId',
+    ''
+  );
+  const relevantReaction = reactionTextBook?.find(
+    (reaction) => reaction.id === activeReactionId
+  );
+  const activeSearchBar = false;
+  return (
+    <Section
+      title={
+        <ChangingInput
+          title={
+            relevantReaction
+              ? 'Reaction: ' + relevantReaction.name
+              : 'Reaction Lookup'
+          }
+          icon="search"
+          onChange={(keyword) =>
+            setActiveReactionId(
+              reactionTextBook.find((reaction) =>
+                reaction.name
+                  .toLowerCase()
+                  .startsWith(keyword.toLocaleLowerCase())
+              )?.id || '')}
+        />
+      }
+      fill
+      scrollable>
+      {relevantReaction && (
+        <>
+          <Box mb={'0.5em'}>{relevantReaction.description}</Box>
+          <Box mb={'0.5em'}>{'Relevant Factors:'}</Box>
+          <LabeledList>
+            {relevantReaction.factors.map((factor) => (
+              <LabeledList.Item
+                key={`${relevantReaction.id}_${factor.factor_name}`}
+                label={
+                  factor.factor_type === 'gas' && factor.factor_id ? (
+                    <Flex>
+                      <Flex.Item
+                        className="AtmosHandbook__link-content"
+                        shrink
+                        onClick={() =>
+                          setActiveGasId(String(factor.factor_id))}>
+                        {factor.factor_name + ':'}
+                      </Flex.Item>
+                    </Flex>
+                  ) : factor.tooltip ? (
```

Comment:
This ternary is really hard to read, can you at least put it in more parentheses?